test/modern_bpf/test_suites/syscall_enter_suite/signalfd4_e.cpp

```diff
@@ -0,0 +1,59 @@
+#include "../../event_class/event_class.h"
+
+#ifdef __NR_signalfd4
+
+#include <sys/signalfd.h>
+
+TEST(SyscallEnter, signalfd4E)
+{
+
+	/* Please note:
+	 * the syscall `signalfd4` is mapped to `PPME_SYSCALL_SIGNALFD_E` event
+	 * like `signalfd`. The same BPF program will be used for both the syscalls.
+	 */
+
+	auto evt_test = new event_test(__NR_signalfd4, ENTER_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	/* `mask` and `flags` are not catched BPF side. */
+	int32_t mock_fd = -1;
+	sigset_t mask = {0};
+	int flags = 7;
+	assert_syscall_state(SYSCALL_FAILURE, "signalfd4", syscall(__NR_signalfd4, mock_fd, &mask, flags));
```

Comment:
This also relates to the `signalfd` BPF program. Looks like there is a similar pattern like described [here](https://github.com/falcosecurity/libs/pull/516/commits/149c181dbf735d61f1ce33a7b0462786ddbb13a7#r935482912).

According to the man page, the system call includes an additional argument, `size_t sizemask` which is transparently handled by the glibc wrapper. For `signalfd4` it is the third argument (where `flags` is passed). I would assume that for calling `signalfd` and `signalfd4` directly, this parameter needs to be added here as well. WDYT?

(I think for now, this is  a blind spot because the BPF programs do not catch those arguments)

For reference, [syscall definition in the Linux kernel](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/fs/signalfd.c#n302).

Reply:
This is really a great catch thank you for your accuracy!! :heart_eyes: